### PR TITLE
migrating env['path'] search logic

### PR DESCRIPTION
**Pull request by heckj**
- migrating from Commands.UserToolchain to Utility.Process
 - migrated tests to match logic move

These changes are in service to broader changes to normalize testing for needed tools (git, xcrun, etc) and error reporting (pulled out from pending PR: https://github.com/apple/swift-package-manager/pull/994)

### CODE REVIEW

File: Sources/Utility/Process.swift [hunk 1]
```diff
@@ -373,6 +373,98 @@ extension Process {
     }
 }
 
+// MARK: utility function for Process
+
+// NOTE FOR PR: follow on from bugfix PR comments https://github.com/apple/swift-package-manager/pull/921/files
```
Comment (aciidb0mb3r) [inline Sources/Utility/Process.swift]: Please remove the commented code.

File: Sources/Utility/Process.swift [hunk 2]
```diff
@@ -373,6 +373,98 @@ extension Process {
...
+    //(type of environmentPath is String?)
+    //let paths = oldPath.components(separatedBy: ":")
+        // if the name starts with a '/', return it explicitly if it exists and can be executed
+
+        // iterate through paths
+            // if exists && can_execute, then return path to it
+            // else throw ProgramSearchError.couldNotFindProgram
+//}
+
+/// Create a list of AbsolutePath search paths from a string, such as the PATH environment variable.
```
Comment (aciidb0mb3r) [inline Sources/Utility/Process.swift]: These two methods should not be in `Process.swift`, lets put them in misc.swift for now.

File: Sources/Utility/Process.swift (hunk 1 above)
Comment (heckj) [inline Sources/Utility/Process.swift]: ah, yep - will do - apologies, missed that in shifting this set

File: Sources/Utility/Process.swift (hunk 2 above)
Comment (heckj) [inline Sources/Utility/Process.swift]: Sounds good, I'll shift them - wasn't entirely sure where to move them within the Utility package.

**Comment by heckj** [comment]
@aciidb0mb3r migrated to `Utility/misc.swift` and rebased back down to a single commit

File: Tests/UtilityTests/miscTests.swift [hunk 3]
```diff
@@ -11,6 +11,7 @@
 import XCTest
 import TestSupport
 @testable import Utility
+@testable import Basic
```
Comment (aciidb0mb3r) [inline Tests/UtilityTests/miscTests.swift]: We shouldn't testable import `Basic` here

Comment (heckj) [inline Tests/UtilityTests/miscTests.swift]: Will update, thanks

**Comment by heckj** [comment]
@aciidb0mb3r updated to remove the extraneous `@testable` and rebased to a single commit

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
Doesn't compile on linux
```
--- bootstrap: note: building stage1
--- bootstrap: note: building runtime v3 target: PackageDescription:
--- bootstrap: note: building runtime v4 target: PackageDescription4:
--- bootstrap: note: building self-hosted 'swift-build': env SWIFT_EXEC=/build/debug/swiftc SWIFT_BUILD_PATH=/build /build/debug/swift-build-stage1 -Xlinker -rpath -Xlinker $ORIGIN/../lib/swift/linux --build-tests
Compile Swift Module 'CommandsTests' (4 sources)
/swiftpm/Tests/CommandsTests/XCTestManifests.swift:19:18: error: use of unresolved identifier 'UserToolchainTests'
        testCase(UserToolchainTests.allTests),
                 ^~~~~~~~~~~~~~~~~~
<unknown>:0: error: build had 1 command failures
error: terminated(1): /build/debug/swift-build-tool -f /build/debug.yaml test
--- bootstrap: error: build failed with exit status 1
```

**Comment by heckj** [comment]
@aciidb0mb3r thanks, I'll go hunt it down - apologies for missing this earlier

**Comment by aciidb0mb3r** [comment]
no issues, you can use a VM for building on linux. I use Ubuntu 16.04 but 14.04, 16.04 and 16.10 are officially supported.

**Comment by heckj** [comment]
@aciidb0mb3r another learning experience :-) I missed removing the file from the Tests/CommandsTests/XCTestManifests.swift when I migrated those test cases with the code - only used on Linux. The docker bits you pointed me at a few weeks ago helped me resolve it pretty quickly, but still working on how to make that a regular part of my dev setup/process. Anyway - PR updated, building and passing in linux (latest master snapshot too) as well as locally.

**Comment by aciidb0mb3r** [comment]
Perfect, thanks!

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**